### ISSUE TITLE
Version 2.5.3 not syncing to Firefox browser extension

### ISSUE THREAD

**Issue by NoWay426** [keepassxreboot/keepassxc#4220]
## Expected Behavior

Upon updating desktop app to version 2.5.3, failed integration with KeePassXC-Browser Version: 1.5.4 (Firefox 72.0.2 64-bit) which had been previously working with no issues

## Current Behavior

Desktop app opens automatically when Firefox opens, prompting password. Database opens successfully. Firefox browser extension displays "Message encryption failed. Is KeePassXC running?" (yes). Hit "Reload" on browser extension, Desktop App closes unexpectedly, browser extension displays "Key exchange was not successful."

## Possible Solutions

Tried the following:

1) Restart of computer upon updating KeePassXC desktop app to v 2.5.3
2) Remove/reconnect database in browser extension settings (will not reconnect)
3) Disable/re-enable browser integration in desktop app
4) Uninstall/reinstall browser extension
5) Uninstall/reinstall desktop app
6) Make sure browser integration is enabled for Firefox and "Use a proxy extension between  KeePassXC and browser extension" is disabled
7) Disable ESET anti-virus software and security/privacy browser extensions (cookie auto-delete, Disconnect, AdBlock Plus)
8) Create a new database, import old pswrds, and try to sync browser extension to new db

Nothing has solved my issue so far. Assistance appreciated.

## Steps to Reproduce

1. Update to KeePassXC 2.5.3
2. Restart computer
3. See if Firefox browser integration still works

## Context

I can no longer use the KeePassXC browser extension, a time-consuming inconvenience. 

## Debug Info

KeePassXC - Version 2.5.3
Revision: f8c962b
Installed via MSI Installer (KeePassXC-2.5.3-Win64)

KeePassXC-Browser Version: 1.5.4

Qt 5.13.2
Debugging mode is disabled.

Operating system: Windows 10 (10.0)
CPU architecture: x86_64
Kernel: winnt 10.0.17763

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey

Cryptographic libraries:
 libgcrypt 1.8.5

**Comment by NoWay426** [comment]
Forgot to mention, running KeePassXC-Browser Version: 1.5.4

**Comment by greg0r** [comment]
Yes, same here with Firefox 72.0.1 on a Ubuntu System. Upgraded yesterday to 2.5.3 no sync since then.

**Comment by varjolintu** [comment]
With 2.5.2 everything still works? No changes has been made between 2.5.2 and 2.5.3 that should affect the connection.

I'd suggest you both take a look at the (Troubleshooting guide)[https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide].

**Comment by droidmonkey** [comment]
Why are you disabling the proxy

**Comment by NoWay426** [comment]
Ahh I see the issue now. @droidmonkey For previous versions of KeePassXC, I had to disable the proxy in order for browser integration to work. I got that suggestion from the comments in this article:

https://davidebove.com/blog/2017/10/11/connecting-keepassxc-and-firefox-for-automated-logins/

For version 2.5.3 I was having the opposite problem, browser integration would not work without that setting enabled. Everything running smoothly now.

@greg0r try enabling "Use a proxy extension between KeePassXC and browser extension" in Tools > Settings > Browser Integration > Advanced Settings
